Assets/VRTK/Scripts/VRTK_RealMovement.cs

```diff
@@ -0,0 +1,350 @@
+﻿// Touchpad Walking|Scripts|0140
+namespace VRTK
+{
+    using UnityEngine;
+    using System.Collections.Generic;
+    using System;
+
+    [RequireComponent(typeof(VRTK_PlayerPresence))]
+
+    /// <summary>
+    /// Real Movement allows you to move the play area by calculating the y-movement of your controllers and propelling you the more they are moving.  This simulates walking in game by walking in real life.
+    /// </summary>
+    /// <remarks>
+    /// It is recommend you combine this with the VRTK_HeightAdjustTeleport
+    /// </remarks>
+    /// <example>
+    /// `VRTK/Examples/042_CameraRig_RealMovement` shows how the user can move and traverse colliders.
+    /// </example>
+    public class VRTK_RealMovement : MonoBehaviour
+    {
+
+        /// <summary>
+        /// Options for testing if a play space fall is valid
+        /// </summary>
+        /// <param name="HeadsetAndControllers">Track both headset and controllers for movement calculations.</param>
+        /// <param name="ControllersOnly">Track only the controllers for movement calculations.</param>
+        /// <param name="HeadsetOnly">Track only headset for movement caluclations.</param>
+        public enum ControlOptions
+        {
+            HeadsetAndControllers,
+            ControllersOnly,
+            HeadsetOnly,
+        }
+        
+        public enum DirectionalMethod
+        {
+            Gaze,
+            DumbDecoupling,
+            SmartDecoupling
+        }
+
+        public bool LeftController
+        {
+            get { return leftController; }
+            set
+            {
+                leftController = value;
+                SetControllerListeners(controllerLeftHand);
+            }
+        }
+
+        public bool RightController
+        {
+            get { return rightController; }
+            set
+            {
+                rightController = value;
+                SetControllerListeners(controllerRightHand);
+            }
+        }
+
+        [Tooltip("If this is checked then the left controller touchpad will be enabled to move the play area. It can also be toggled at runtime.")]
+        [SerializeField]
+        private bool leftController = true;
+        [Tooltip("If this is checked then the right controller touchpad will be enabled to move the play area. It can also be toggled at runtime.")]
+        [SerializeField]
+        private bool rightController = true;
+
+        [Tooltip("Select which trackables are used to determine movement.")]
+        [SerializeField]
+        private ControlOptions m_controlOptions = ControlOptions.HeadsetAndControllers;
```

Comment:
the style of the toolkit doesn't use snake case variables. only use camel case. e.g. `controlOptions`